src/libcore/fmt/mod.rs

```diff
@@ -274,14 +274,119 @@ impl<'a> ArgumentV1<'a> {
 
     #[doc(hidden)]
     #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
-    pub fn new<'b, T>(x: &'b T, f: fn(&T, &mut Formatter<'_>) -> Result) -> ArgumentV1<'b> {
-        unsafe { ArgumentV1 { formatter: mem::transmute(f), value: mem::transmute(x) } }
+    #[cfg(bootstrap)]
+    pub fn new<'b, T>(x: &'b T, formatter: fn(&T, &mut Formatter<'_>) -> Result) -> ArgumentV1<'b> {
+        unsafe { ArgumentV1 { value: mem::transmute(x), formatter: mem::transmute(formatter) } }
+    }
+
+    #[doc(hidden)]
+    #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
+    pub fn new_display<T: Display>(x: &T) -> ArgumentV1<'_> {
+        unsafe {
```

Comment:
Since you're adding new code, please add `// SAFETY` comments on each `unsafe { ... }`. (We're telling tidy to ignore it atm for this file but we shouldn't regress things.)

Reply:
This is very much an in-progress PR and purely posted for the perf (hence `r? @ghost`), but I'll definitely do so in the future.

Reply:
Sounds great, thanks! 👍